doschema/jsvalidation.py

```diff
@@ -0,0 +1,320 @@
+            field_type = curr_schema['type']
+
+            self.fields_types_dict[curr_field] = field = FieldToAdd(
+                schema_index=self.uri,
+                field_tuple=curr_field,
+                field_type=field_type
+            )
+
+            if curr_schema['type'] == 'array':
+                self._validate_array(
+                    curr_schema['items'],
+                    curr_field + ('items',)
+                )
+        if "enum" in curr_schema:
+            enum_type = self._validate_enum(
+                curr_schema['enum'],
+                curr_field
+            )
+            if field is not None and field.field_type != enum_type:
+                err_msg = """Conflicting type for field {0} previously found
+                            in schema {1} enum with type {2},
+                            found next with type {3}."""
+                raise JSONSchemaCompatibilityError(
+                    self.uri,
+                    err_msg.format(
+                        self._make_json_pointer(curr_field),
+                        self.uri,
+                        field.schema_index,
+                        field.field_type
+                    )
+                )
+            self.fields_types_dict[curr_field] = field = FieldToAdd(
+                schema_index=self.uri,
+                field_tuple=curr_field,
+                field_type=enum_type
+            )
+
+        if 'properties' in curr_schema:
+            if field and field.field_type != "object":
+                err_msg = """Conflicting type for field {0} previously found
+                            in schema {1} with type {2}, found next in {3}
+                            where "dependencies" imply type "object"."""
+                raise JSONSchemaCompatibilityError(
+                    field.schema_index,
+                    self.uri,
+                    err_msg.format(
+                        self._make_json_pointer(curr_field),
+                        field.schema_index,
+                        field.field_type,
+                        self._make_json_pointer(field.field_tuple)
+                    )
+                )
+            self.fields_types_dict[curr_field] = field = FieldToAdd(
+                schema_index=self.uri,
+                field_tuple=curr_field,
+                field_type='object'
+            )
+            for prop in curr_schema['properties']:
+                self._validate_root(
+                    curr_schema['properties'][prop],
+                    curr_field + (prop,)
+                )
+
+        if "dependencies" in curr_schema:
+            if field and field.field_type != "object":
+                err_msg = """Conflicting type for field {0} previously found
+                            in schema {1} with type {2}, found next in {3}
+                            where "dependencies" imply type "object"."""
+                raise JSONSchemaCompatibilityError(
+                    self.uri,
```

Comment:
Bug after refactoring? `self.uri` is not the previous schema. The previous schema can be left as None.